File: app/src/main/java/eu/indiewalkabout/mathbrainer/statistics/ResultsManager.java
package eu.indiewalkabout.mathbrainer.statistics;


/**
 * Singletons class for storing and retrieving match data
 *
 */
public class ResultsManager {

    // singleton instantiation
    private static final Object LOCK = new Object();
    private static ResultsManager resultsManagerSingleton;


    public static ResultsManager getInstance() {
        if(resultsManagerSingleton == null){
            synchronized (LOCK){
                resultsManagerSingleton = new ResultsManager();
            }
        }
        return resultsManagerSingleton;
    }


}

File: app/src/main/java/eu/indiewalkabout/mathbrainer/core/notused/ResultsManager.kt
package eu.indiewalkabout.mathbrainer.core.notused

// Singletons class for storing and retrieving match data
object ResultsManager {
    // singleton instantiation
    private val LOCK = Any()
    private var resultsManagerSingleton: ResultsManager? = null
    val instance: ResultsManager?
        get() {
            if (resultsManagerSingleton == null) {
                synchronized(LOCK) { resultsManagerSingleton = ResultsManager }
            }
            return resultsManagerSingleton
        }
}
